core/src/main/java/org/opensearch/sql/analysis/ExpressionAnalyzer.java

```diff
@@ -148,6 +156,55 @@ public Expression visitNot(Not node, AnalysisContext context) {
     return dsl.not(node.getExpression().accept(this, context));
   }
 
+  @Override
+  public Expression visitRelevanceFunction(RelevanceFunction node, AnalysisContext context) {
+    FunctionName functionName = FunctionName.of(node.getFuncName());
+    TypeEnvironment typeEnv = context.peek();
+
+    List<Expression> arguments =
+        node.getFuncArgs().stream()
+            .map(unresolvedExpression -> analyze(unresolvedExpression, context))
+            .collect(Collectors.toList());
+
+    // Ensure each field is valid
+    switch(node.getType()) {
+      case NO_FIELD_FUNCTION:
+        Predicate<String> filter_predicate = Pattern
+            .compile("[^*]+:.+")
+            .asPredicate();
+
+        arguments.stream().filter(arg -> ((NamedArgumentExpression)arg).getArgName().equals("query"))
+            .forEach(query -> Arrays.asList(StringUtils.unquoteText(((NamedArgumentExpression) query)
+                    .getValue().toString()).split("\\s+")).stream().filter(filter_predicate)
+                .collect(Collectors.toList()).stream().map(
+                    p -> p.split(":")[0]).collect(Collectors.toList())
```

Comment:
You are splitting by whitespace, matching by field:query pattern, then splitting out the field.  
I would suggest writing a pattern to do all of this in one, and then returning the matchResult as one. 
```
String[] matches = Pattern.compile("your regex here")
                          .matcher("string to search from here")
                          .results()
                          .map(MatchResult::group)
                          .toArray(String[]::new);
                    // or .collect(Collectors.toList())
```
see: https://stackoverflow.com/questions/6020384/create-array-of-regex-matches

Reply:
That being said, do we really want to maintain a pattern matcher for query expressions that may break for any changes to how OpenSearch handles query expressions?  Do we know if they handle tabs, or carriage returns?  What about queries with whitespace?  

I feel like this isn't a good road to go down.  I would prefer to not check `NO_FIELD_FUNCTION` relevance functions.  Especially since these functions are just used (for now) to cover backwards compatibility.